File: frontend/types/prediction.ts
export interface ClassScore {
  class_id: string;
  label: string;
  confidence: number;
  risk_level: string;
}

export interface PredictionResponse {
  predicted_class: string;
  class_label: string;
  confidence: number;
  risk_level: string;
  all_predictions: ClassScore[];
}

export interface HealthStatus {
  status: string;
  model_loaded: boolean;
  device: string;
}

export interface CancerInfo {
  label: string;
  scientific_name: string;
  risk_level: string;
  color: string;
  description: string;
  prevalence: string;
  symptoms: string[];
  treatment: string;
  when_to_see_doctor: string;
}

File: backend/schemas/prediction.py
from pydantic import BaseModel
from typing import List

class ClassScore(BaseModel):
    class_id: str
    label: str
    confidence: float
    risk_level: str

class PredictionResponse(BaseModel):
    predicted_class: str
    class_label: str
    confidence: float
    risk_level: str
    all_predictions: List[ClassScore]

class HealthStatus(BaseModel):
    status: str
    model_loaded: bool
    device: str
